tools: move webcrypto into no-restricted-properties

ZihongQu opened the pull request:
Since `eslint` fixed https://github.com/eslint/eslint/issues/16412 in `v8.56.0` [here](https://github.com/eslint/eslint/releases/tag/v8.56.0) and we are on `eslint` `v8.57.0` so that we can take advantage of `no-restricted-properties` rule for `webcrypto`.

This is my first pull request to node.js. Please let me know if there is anything to fix.

<!--
Before submitting a pull request, please read:

- the CONTRIBUTING guide at https://github.com/nodejs/node/blob/HEAD/CONTRIBUTING.md
- the commit message formatting guidelines at
  https://github.com/nodejs/node/blob/HEAD/doc/contributing/pull-requests.md#commit-message-guidelines

For code changes:
1. Include tests for any bug fixes or new features.
2. Update documentation if relevant.
3. Ensure that `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes.

If you believe this PR should be highlighted in the Node.js CHANGELOG
please add the `notable-change` label.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
-->

aduh95 commented on .eslintrc.js:
```diff
@@ -187,6 +187,10 @@ module.exports = {
         message: 'Use `assert.notStrictEqual()` rather than `assert.notEqual()`.',
       },
+      {
+        property: 'webcrypto',
```
```suggestion
        object: 'crypto',
        property: 'webcrypto',
```

jakecastelli replied:
Just checked this PR locally - looks like with `object: crypto` it will prevent cases like

```js
const { webcrypto } = crypto;
```

but it does not prevent from

```js
const { webcrypto } = require('crypto').

// or 

const { subtle } = require('node:crypto').webcrypto;
```

Therefore, I think this lint rule should be ok.

aduh95 commented on test/parallel/test-global-webcrypto.js:
```diff
@@ -7,7 +7,7 @@ if (!common.hasCrypto)
 const crypto = require('crypto');
 
-/* eslint-disable no-restricted-syntax */
+/* eslint-disable no-restricted-properties, no-restricted-syntax */
```
Do we still need to disable no-restricted-syntax?
```suggestion
/* eslint-disable no-restricted-properties */
```

jakecastelli replied:
Looks like the `no-restricted-syntax` error comes from `test/.eslintrc.yaml` [here](https://github.com/nodejs/node/blob/main/test/.eslintrc.yaml#L53) I think the `no-restricted-properties` should also be added into `test/.eslintrc.yaml` in order to keep the rule consistent.

ZihongQu replied:
Adding `no-restricted-properties` into `test/.eslintrc.yaml` will result in more linting errors in other test files. I have added a `TODO` which I will try to fix in a separate PR.

aduh95 commented:
Can you rebase please? With the move to flat config, there are some conflicts to solve.

ZihongQu commented:
> Can you rebase please? With the move to flat config, there are some conflicts to solve. 

Hi @aduh95, I have rebased my branch. Could you please take a look again and rerun CI for me? Thanks. 

I have also removed my `TODO` as the upgrade of `Eslint` from `v8` to `v9` does not cause errors in the test files anymore.

aduh95 commented:
The linter is failing (also it looks like the TODO is still there?)

ZihongQu commented:
> The linter is failing (also it looks like the TODO is still there?)

Oh sorry. I accidentally dropped one of my commit during interactive rebase. Thank you for spotting it! It should be fixed by now.

ZihongQu commented:
Hi @aduh95, I am still seeing 4 failing CI jobs (feels like flaky tests) could you help me take a look, please? Thanks a lot ! 👀
